File: June 2025/10.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int countStrings(string &s) {
        unordered_map<char, int> freq;
        for (char ch : s) {
            freq[ch]++;
        }

        int n = s.size();
        int total_swaps = (n * (n - 1)) / 2; // All possible index swaps

        // Count duplicate swaps where characters are identical
        int duplicate_count = 0;
        for (auto &entry : freq) {
            int val = entry.second;
            duplicate_count += (val * (val - 1)) / 2;
        }

        return total_swaps - duplicate_count + (duplicate_count > 0 ? 1 : 0);
    }
};

File: June 2025/10.py
class Solution:
    def countStrings(self, s): 
        freq = {}
        for char in s:
            freq[char] = freq.get(char, 0) + 1

        n = len(s)
        total_swaps = n * (n - 1) // 2  # All possible index swaps

        # Count duplicate swaps where the swapped characters are identical
        duplicate_count = sum((val * (val - 1)) // 2 for val in freq.values())

        return total_swaps - duplicate_count + (1 if duplicate_count > 0 else 0)
